test/gtest/ucp/test_ucp_tag_match.cc

```diff
@@ -541,6 +450,143 @@ UCS_TEST_P(test_ucp_tag_match, sync_send_unexp) {
+                                              send_callback);
+            ASSERT_TRUE(!UCS_PTR_IS_ERR(sreq1));
+
+            sreq2 = (request*)ucp_tag_send_nb(sender2.ep(), &sendbuf2[0],
+                                              sendbuf2.size(), DATATYPE, 2,
+                                              send_callback);
+            ASSERT_TRUE(!UCS_PTR_IS_ERR(sreq2));
+
+            /* In unexpected mode, we progress all to put the messages on the
+             *  unexpected queue
+             */
+            if (!is_exp) {
+                short_progress_loop();
+            }
+
+            /* Two receives with any tag */
+
+            request *rreq1, *rreq2;
+
+            rreq1 = recv_nb(&recvbuf1[0], recvbuf1.size(), DATATYPE, 0, 0);
+            ASSERT_TRUE(!UCS_PTR_IS_ERR(rreq1));
+
+            rreq2 = recv_nb(&recvbuf2[0], recvbuf2.size(), DATATYPE, 0, 0);
+            ASSERT_TRUE(!UCS_PTR_IS_ERR(rreq2));
+
+            /* Wait for receives */
+            wait(rreq1);
+            wait(rreq2);
+
+            short_progress_loop();
+
+            /* Release sends */
+            if (sreq1 != NULL) {
+                wait(sreq1);
+                EXPECT_TRUE(sreq1->completed);
+                request_free(sreq1);
+            }
+            if (sreq2 != NULL) {
+                wait(sreq2);
+                EXPECT_TRUE(sreq2->completed);
+                request_free(sreq2);
+            }
+
+            /* Receives should be completed with correct length */
+            ASSERT_TRUE(rreq1->completed);
+            ASSERT_TRUE(rreq2->completed);
+
+            EXPECT_EQ(size, rreq1->info.length);
+            EXPECT_EQ(size, rreq2->info.length);
+
+            /* The order may be any, but the messages have to be received correctly */
+            if (rreq1->info.sender_tag == 1u) {
+                ASSERT_EQ(2u, rreq2->info.sender_tag);
+                EXPECT_EQ(sendbuf1, recvbuf1);
+                EXPECT_EQ(sendbuf2, recvbuf2);
+            } else {
+                ASSERT_EQ(2u, rreq1->info.sender_tag);
+                ASSERT_EQ(1u, rreq2->info.sender_tag);
+                EXPECT_EQ(sendbuf2, recvbuf1);
+                EXPECT_EQ(sendbuf1, recvbuf2);
+            }
+
+            request_free(rreq1);
+            request_free(rreq2);
+        }
+    }
+};
+
+UCP_INSTANTIATE_TEST_CASE(ucp_tag_send_nb_recv_medium_wildcard)
+
+UCS_TEST_P(ucp_tag_send_nb_recv_medium_wildcard, test, "RNDV_THRESH=inf")
+{
+    test();
+}
+
+class ucp_tag_wcd_hw_proto_small_rxq : public ucp_tag_send_nb_recv_medium_wildcard
+{
+public:
+    virtual void init()
+    {
+        enable_tag_mp_offload();
+        ASSERT_TRUE(use_proto());
+        modify_config("PROTO_ENABLE", "y", SETENV_IF_NOT_EXIST);
+        modify_config("RNDV_THRESH", "inf");
+        modify_config("TM_THRESH", "1");
+        modify_config("RC_RX_QUEUE_LEN", "1024", SETENV_IF_NOT_EXIST);
+        test_ucp_tag::init();
```

Comment:
Why not call here `test_ucp_tag_match::init()` instead?
Then can remove `PROTO_ENABLE` and `TM_THRESH` from here.